pkg/apis/work/v1alpha1/binding_types.go

```diff
@@ -66,6 +66,34 @@ type ObjectReference struct {
 	ResourceVersion string `json:"resourceVersion,omitempty"`
 }
 
+// ReplicaRequirements represents the requirements required by each replica.
+type ReplicaRequirements struct {
+	// NodeClaim represents the node claim NodeAffinity, NodeSelector and Tolerations required by each replica.
+	// +optional
+	NodeClaim *NodeClaim `json:"nodeClaim,omitempty"`
+
+	// ResourceRequest represents the resources required by each replica.
+	// +optional
+	ResourceRequest corev1.ResourceList `json:"resourceRequest,omitempty"`
+}
+
+// NodeClaim represents the node claim NodeAffinity, NodeSelector and Tolerations required by each replica.
+type NodeClaim struct {
+	// A node selector represents the union of the results of one or more label queries over a set of
+	// nodes; that is, it represents the OR of the selectors represented by the node selector terms.
+	// Note that only PodSpec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+	// is included here because it has a hard limit on pod scheduling.
+	// +optional
+	NodeAffinity *corev1.NodeSelector `json:"nodeAffinity,omitempty"`
```

Comment:
nit: shall we use another name e.g. `hardNodeAffinity`? Since it's different struct here.

Reply:
Good idea.